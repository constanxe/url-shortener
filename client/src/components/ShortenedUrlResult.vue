<template>
  <div v-if="shortenedUrl">
    Shortened URL: <a target="_blank" :href="shortenedUrl">{{shortenedUrl}}</a>
    &nbsp;<button @click="copyShortenedUrl()">Copy</button>
  </div>
</template>

<script>
export default {
  name: 'ShortenedUrlResult',
  props: {
    shortenedUrl: String
  },
  methods: {
    copyShortenedUrl() {
      navigator.clipboard.writeText(this.shortenedUrl);
    },
  }
}
</script>
