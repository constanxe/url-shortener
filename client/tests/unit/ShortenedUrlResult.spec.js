import {shallowMount} from '@vue/test-utils'
import ShortenedUrlResult from '@/components/ShortenedUrlResult.vue'

Object.assign(navigator, {
  clipboard: {writeText: jest.fn()}
})

describe('ShortenedUrlResult.vue', () => {
  let wrapper
  let shortenedUrl = 'http://localhost:3000/qwe123'

  const getWrapper = () => shallowMount(ShortenedUrlResult, {
    propsData: {shortenedUrl}
  })

  beforeEach(() => wrapper = getWrapper())

  it('copies props.shortenedUrl when button clicked', () => {
    wrapper.find('button').trigger('click')

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith(shortenedUrl)
  })

  describe('displays content if props.shortenedUrl passed', () => {
    const wrapperMatchesSnapshot = () => expect(wrapper).toMatchSnapshot()

    it('renders props.shortenedUrl when passed', () => {
      expect(wrapper.find(`a[href="${shortenedUrl}"]`).exists()).toBeTruthy()
      expect(wrapper.text()).toMatch('Shortened URL: ' + shortenedUrl)
      wrapperMatchesSnapshot()
    })

    it('does not show content when props.shortenedUrl empty', () => {
      shortenedUrl = ''
      wrapper = getWrapper()

      expect(wrapper.find('a').exists()).toBeFalsy()
      expect(wrapper.text()).toEqual(shortenedUrl)
      wrapperMatchesSnapshot()
    })
  })
})
