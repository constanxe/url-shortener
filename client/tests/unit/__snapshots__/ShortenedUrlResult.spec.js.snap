// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortenedUrlResult.vue displays content if props.shortenedUrl passed does not show content when props.shortenedUrl empty 1`] = ``;

exports[`ShortenedUrlResult.vue displays content if props.shortenedUrl passed renders props.shortenedUrl when passed 1`] = `
<div><b>Shortened URL: <a target="_blank" href="http://localhost:3000/qwe123">http://localhost:3000/qwe123</a></b>
  &nbsp;<button>Copy</button></div>
`;
